### PR TITLE
[EOC-417] Proposal add notification additional info when leaving saved data

### PR DESCRIPTION
**Pull request by adamklepacz**
End_of_Coffee_-_tradsack_your_office_inventory
End_of_Coffee_-_track_your_office_inventory

### CODE REVIEW

File: client/src/common/utils/events.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export const attachBeforeUnloadEvent = callback =>
+  window.addEventListener('beforeunload', callback);
```
Comment (alemikolo) [inline client/src/common/utils/events.js]: hmm, if we use sth like this:
```
export const attachBeforeUnloadEvent = (callback = handleWindowBeforeUnload) =>
  window.addEventListener('beforeunload', callback);
```
there will be node need to import `handleWindowBeforeUnload` in ever file where we want to add/remove onBeforeUnload event, but we will be able to pass different callback;

File: client/src/modules/cohort/components/CohortDescription.jsx [hunk 2]
```diff
@@ -43,10 +48,24 @@ class CohortDescription extends PureComponent {
     const {
       details: { description: previousDescription }
     } = previousProps;
+    const { descriptionInputValue } = this.state;
+    const dataHasChanged = previousDescription !== descriptionInputValue;
```
Comment (alemikolo) [inline client/src/modules/cohort/components/CohortDescription.jsx]: inputValueHasChanged?

File: client/src/modules/cohort/components/CohortName.jsx [hunk 3]
```diff
@@ -50,11 +55,25 @@ class CohortName extends PureComponent {
     const {
       details: { name: previousName }
     } = previousProps;
-    const hasNameBeenChanged = name !== previousName;
+    const { nameInputValue } = this.state;
+    const nameHasChanged = name !== previousName;
+    const dataHasChanged = previousName !== nameInputValue;
```
Comment (alemikolo) [inline client/src/modules/cohort/components/CohortName.jsx]: inputValueHasChanged?

File: client/src/common/utils/events.js (hunk 1 above)
Comment (adamklepacz) [reply client/src/common/utils/events.js]: @Ciunkos  @marekrozmus  what do you think?

File: client/src/modules/cohort/components/CohortDescription.jsx (hunk 2 above)
Comment (adamklepacz) [reply client/src/modules/cohort/components/CohortDescription.jsx]: I think data has changed is understandable enough.

File: client/src/modules/cohort/components/CohortName.jsx (hunk 3 above)
Comment (adamklepacz) [reply client/src/modules/cohort/components/CohortName.jsx]: I think data has changed is understandable enough.

**Pull request merged**